pkg/controller/upload-controller.go

```diff
@@ -490,37 +332,320 @@ func (c *UploadController) getOrCreateUploadService(pvc *v1.PersistentVolumeClai
 	return service, nil
 }
 
-func (c *UploadController) deleteService(namespace, name string) error {
-	service, err := c.serviceLister.Services(namespace).Get(name)
-	if err != nil {
+func (r *UploadReconciler) deleteService(namespace, serviceName string) error {
+	service := &corev1.Service{}
+	if err := r.Client.Get(context.TODO(), types.NamespacedName{Name: serviceName, Namespace: namespace}, service); err != nil {
 		if k8serrors.IsNotFound(err) {
 			return nil
 		}
-		return errors.Wrap(err, "error getting upload service")
+		return err
 	}
 
 	if service.DeletionTimestamp == nil {
-		err = c.client.CoreV1().Services(namespace).Delete(name, &metav1.DeleteOptions{})
-		if err != nil {
-			if k8serrors.IsNotFound(err) {
-				return nil
-			}
+		if err := r.Client.Delete(context.TODO(), service); IgnoreNotFound(err) != nil {
 			return errors.Wrap(err, "error deleting upload service")
 		}
 	}
 
 	return nil
 }
 
-// returns the CloneRequest string which contains the pvc name (and namespace) from which we want to clone the image.
-func (c *UploadController) getCloneRequestSourcePVC(pvc *v1.PersistentVolumeClaim) (*v1.PersistentVolumeClaim, error) {
-	exists, namespace, name := ParseCloneRequestAnnotation(pvc)
-	if !exists {
-		return nil, errors.New("error parsing clone request annotation")
+// CreateUploadService creates upload service service manifest and sends to server
+func (r *UploadReconciler) CreateUploadService(name string, pvc *v1.PersistentVolumeClaim) (*v1.Service, error) {
```

Comment:
why is this public?

Reply:
Not sure, no reason for it to be, fixed.